manifests/init.pp

```diff
@@ -38,7 +38,7 @@
   Array[String[1]] $jails = ['ssh', 'ssh-ddos'],
   Integer[0] $maxretry = 3,
   Array $whitelist = ['127.0.0.1/8', '192.168.56.0/24'],
-  $custom_jails = undef,
+  Hash $custom_jails = lookup('fail2ban::custom_jails', Hash, 'deep', {}),
```

Comment:
I like a top level parameter for this, but I'd prefer to rely on users to set up their own hiera hashing. So `Hash[String, Hash] $custom_jails = {}` is probably sufficient